runtime/tests/checker/interface_test.go

```diff
@@ -2689,3 +2689,1069 @@ func TestCheckBadStructInterface(t *testing.T) {
+	})
+}
+
+func TestCheckInterfaceTypeDefinitionInheritance(t *testing.T) {
+
+	t.Parallel()
+
+	t.Run("type requirement", func(t *testing.T) {
+
+		t.Parallel()
+
+		_, err := ParseAndCheck(t, `
+            contract interface A {
+                struct Nested {
+                    pub fun test(): Int {
+                        return 3
+                    }
+                }
+            }
+
+            contract interface B: A {}
+
+            contract interface C: B {}
+
+            contract X: C {
+                struct Nested {
+                    pub fun test(): Int {
+                        return 3
+                    }
+                }
+            }
+        `)
+
+		require.NoError(t, err)
+	})
+
+	t.Run("type requirement negative", func(t *testing.T) {
+
+		t.Parallel()
+
+		_, err := ParseAndCheck(t, `
+            contract interface A {
+                struct Nested {
+                    pub fun test(): Int {
+                        return 3
+                    }
+                }
+            }
+
+            contract interface B: A {}
+
+            contract interface C: B {}
+
+            contract X: C {}
+        `)
+
+		errs := RequireCheckerErrors(t, err, 1)
+		assert.IsType(t, &sema.ConformanceError{}, errs[0])
+	})
+
+	t.Run("type requirement multiple", func(t *testing.T) {
+
+		t.Parallel()
+
+		_, err := ParseAndCheck(t, `
+            contract interface A {
+                struct ANested {
+                    pub fun test(): Int {
+                        return 3
+                    }
+                }
+            }
+
+            contract interface B {
+                struct BNested {
+                    pub fun test(): Int {
+                        return 4
+                    }
+                }
+            }
+
+            contract interface C: A, B {}
+
+            contract X: C {
+                struct ANested {
+                    pub fun test(): Int {
+                        return 3
+                    }
+                }
+
+                struct BNested {
+                    pub fun test(): Int {
+                        return 3
+                    }
+                }
+            }
+        `)
+
+		require.NoError(t, err)
+	})
+
+	t.Run("type requirement multiple not conforming", func(t *testing.T) {
+
+		t.Parallel()
+
+		_, err := ParseAndCheck(t, `
+            contract interface A {
+                struct ANested {
+                    pub fun test(): Int {
+                        return 3
+                    }
+                }
+            }
+
+            contract interface B {
+                struct BNested {
+                    pub fun test(): Int {
+                        return 4
+                    }
+                }
+            }
+
+            contract interface C: A, B {}
+
+            contract X: C {
+                struct ANested {
+                    pub fun test(): Int {
+                        return 3
```

Comment:
How exactly do these implementations interact with the default implementation check? Do we need to prevent `C` from declaring a different implementation for `ANested.test()`? If so we should add tests for that.

Reply:
There was no change to the behavior for type requirements as well.

Here the conformance-error is not with the default implementations, but rather because of the missing/unimplemented type requirement. I'll improve the asserts to also check for the details of the error.

Reply:
Improved the asserts in c535fb201ed7837b5b9b65bc852a796b801fc8da